unit-tests/utilities/time/test-waiting-on.cpp

```diff
@@ -117,3 +151,30 @@ TEST_CASE( "Not invoked but still notified" )
     REQUIRE( waited_ms > 1990 );
     REQUIRE( waited_ms < 3000 );    // Up to a second buffer
 }
+
+TEST_CASE("Not invoked flush timeout expected")
+{
+    // Emulate some dispatcher
+    typedef std::function< void() > func;
+    auto dispatcher = new std::queue< func >;
+
+    // Push some stuff onto it (not important what)
+    int i = 0;
+    dispatcher->push([&]() { ++i; });
+    dispatcher->push([&]() { ++i; });
+
+    // Add something we'll be waiting on
+    std::condition_variable cv;
+    std::mutex m;
+    utilities::time::waiting_on< bool > invoked(cv, m, false);
+    dispatcher->push([invoked_in_thread = invoked.in_thread()]() { invoked_in_thread.signal(true); });
+
+    auto wait_start = std::chrono::high_resolution_clock::now();
+    invoked.wait_until( std::chrono::seconds( 5 ), [&]() { return invoked; } );
```

Comment:
If we're expecting the timeout then 3, even 2 seconds should be enough?

Reply:
👍